### ISSUE TITLE
macos release not self-contained -- requires llvm installed on the system

### ISSUE THREAD

**Issue by hasenj** [odin-lang/Odin#1013]
## Context

Downloading the macos build from this page:

https://github.com/odin-lang/Odin/releases/tag/dev-2021-06

The assumption here is that someone who just wants to use the language, and not build it from source.

* Operating System: macOS
* Please paste `odin version` output: N/A. See below. Release number mentioned above.

## Expected Behavior

Compiler works out of the box.

## Current Behavior

- The executable does not have the executable permission

    - The user must manually run `chmod u+x odin`
   
- The executable is not self-contained. It requires a specific version of llvm to be installed in a specific location

```
$ odin
dyld: Library not loaded: /usr/local/opt/llvm@11/lib/libLLVM-C.dylib
  Referenced from: /Users/hasenj/odin/odin
  Reason: image not found
Abort trap: 6

$ odin version
dyld: Library not loaded: /usr/local/opt/llvm@11/lib/libLLVM-C.dylib
  Referenced from: /Users/hasenj/odin/odin
  Reason: image not found
Abort trap: 6
```

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. Have a macOS system without llvm installed
2. Download the release and unzip it to a reasonable location, e.g. `~/odin` and make sure it's added to the $PATH variable
3. Invoke the `odin` command

**Comment by Platin21** [comment]
I suspect it would make sense to have a llvmlib directory in which this library goes ideally setup with rpath‘s so that it doesn’t require a fixed position for the library.

**Comment by ctrl-dlahr** [comment]
in case anyone else runs across this issue trying out Odin, you can fix it by doing the following:

Download version of LLVM Odin is using

```zsh
brew install llvm@11
```

add to PATH

```zsh
echo 'export PATH="/usr/local/opt/llvm@11/bin:$PATH"' >> ~/.zshrc
```

add execute permission to Odin

```zsh
chmod +x odin
```

Profit

```zsh
./odin run examples/demo/demo.odin -file
```

**Comment by Nezteb** [comment]
In case anyone is trying this in 2023, I was trying to install Odin in my M2 Macbook with [`asdf`](https://github.com/asdf-vm/asdf) and [`asdf-odin`](https://github.com/jtakakura/asdf-odin).

I got an error saying that a `llvm@11/lib/libLLVM-C.dylib` file was needed. I had the file, but the error said it was `(mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64'))`.

That's when I realized that the [GitHub releases don't currently include arm64 builds, but maybe that'll change someday? 😄](https://github.com/odin-lang/Odin/discussions/2830)

Sooooo, for now I'll just have to compile Odin from source on my arm64 machine.
